### PR TITLE
Add Grunt to compress SMap js files.

### PR DESCRIPTION
**Pull request by linzjax**
_[__NOTE__: Not going into __master__]_

### Proposed changes in this pull request
- Addresses #1244 
- setup vagrant provisioning to include grunt.
- Added gruntserver file to run with ./runserver.
- Setup Gruntfile.js to uglify SMap files as a test
- Sorted third party library files into vendor folders
- Renamed router/routes/router_mixins to SimpleRouter, CreateRoutes, and RouterMixins respectively.
- put datatable and parsley files into their own folders
- added jshint to grunt configuration
- removed all es6 logic from js files and linting.

### When should this PR be merged
- Before SMap is put up on staging.

### Risks
- I'm adding a step to VM set up, as well as a step to deployment. `grunt production` should be run before deploying to demo/prod. `grunt concat` should be run when deploying to staging.

### Follow-up actions
- Double check with adri that this actually works.

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?** 
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: cadasta/core/static/js/rel_tenure.js [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
   $('div#select-party').toggleClass('hidden');
   $('div#new-item').toggleClass('hidden');
-});
-
-$('table#select-list tr').click(function(event) {
-  const target = $(event.target).closest('tr');
-  const relId = target.attr('data-id');
-  target.closest('tbody').find('tr.info').removeClass('info');
-  target.addClass('info');
-  $('input[name="id"]').val(relId);
```
Comment (oliverroick) [inline cadasta/core/static/js/rel_tenure.js]: This works but I don't understand how. We must set the party ID somewhere. Where does this happen? ... just curious.

File: provision/roles/node/tasks/main.yml [hunk 2]
```diff
@@ -37,3 +37,9 @@
   become_user: root
   shell: npm install -g npm@2
   when: node_check.stat.exists == False
+
+- name: Make node_module link
+  become: yes
+  become_user: root
+  file: path=/vagrant/node_modules src=/vagrant/cadasta/core/node_modules state=link follow=yes
+  # when: node_check.stat.exists == False
```
Comment (oliverroick) [inline provision/roles/node/tasks/main.yml]: Can you check back with @amplifi what this does and whether we need it here? If we don't need it then just remove it

Comment (linzjax) [inline provision/roles/node/tasks/main.yml]: Removed the comment.

File: cadasta/core/static/js/rel_tenure.js (hunk 1 above)
Comment (oliverroick) [inline cadasta/core/static/js/rel_tenure.js]: Guess we'll find out when it's on staging then ;)

**Pull request merged**